healthvault-ios-sdk/Classes/Clients/MHVPersonClient.m

```diff
@@ -71,55 +131,215 @@ - (void)getAuthorizedPeopleWithCompletion:(void(^)(NSArray<MHVPersonInfo *> *_Nu
+     {
+         if (error)
+         {
+             completion(nil, error);
+             
+             return;
+         }
+         
+         MHVPersonInfo *personInfo;
+         
+         XReader *reader = [[XReader alloc] initFromString:response.infoXml];
+         
+         if ([reader readStartElementWithName:@"info"])
+         {
+             personInfo = (MHVPersonInfo *)[reader readElement:@"person-info" asClass:[MHVPersonInfo class]];
+         }
+         
+         if (!personInfo)
+         {
+             completion(nil, [NSError error:[NSError MHVUnknownError] withDescription:@"The GetPersonInfo response is invalid."]);
+             
+             return;
+         }
+
+         completion(personInfo, nil);
+     }];
+}
+
+- (void)getAuthorizedRecordsWithRecordIds:(NSArray<NSUUID *> *)recordIds
+                               completion:(void(^)(MHVRecordCollection *_Nullable records, NSError *_Nullable error))completion
+{
+    MHVASSERT_PARAMETER(recordIds);
+    MHVASSERT_PARAMETER(completion);
+    
+    if (!completion)
+    {
+        return;
+    }
+    
+    if (!recordIds)
+    {
+        completion(nil, [NSError MVHRequiredParameterIsNil]);
+        return;
+    }
+    
+    MHVMethod *method = [MHVMethod getAuthorizedRecords];
+    method.parameters = [self bodyForGetAuthorizedRecords:recordIds];
+    
+    [self.connection executeHttpServiceOperation:method
+                                      completion:^(MHVServiceResponse * _Nullable response, NSError * _Nullable error)
+     {
+         if (error)
+         {
+             completion(nil, error);
+             
+             return;
+         }
+         
+         MHVRecordCollection *healthRecords = (MHVRecordCollection *)[XSerializer newFromString:response.infoXml
+                                                                                       withRoot:@"info"
+                                                                                 andElementName:@"record"
+                                                                                        asClass:[MHVRecord class]
+                                                                                  andArrayClass:[MHVRecordCollection class]];
+         
+         if (healthRecords.count != recordIds.count)
+         {
+             MHVLOG(@"Warning: %li records requested, %li results", recordIds.count, healthRecords.count);
+         }
+         
+         completion(healthRecords, nil);
+     }];
+}
+
+#pragma mark - Helpers
+
+- (void)getAuthorizedPeopleWithParameters:(NSString *_Nonnull)parameters
+                               completion:(void(^)(MHVGetAuthorizedPeopleResult *_Nullable authorizedPeople, NSError *_Nullable error))completion
+{
     MHVMethod *method = [MHVMethod getAuthorizedPeople];
-    method.parameters = @"<info><parameters></parameters></info>";
+    method.parameters = parameters;
     
     [self.connection executeHttpServiceOperation:method
```

Comment:
Unfortunately, we can't call this done until we handle the case where there the result is incomplete (moreResults is YES). You are going to have to track down how this actually works.